v2/pkg/protocols/common/interactsh/interactsh.go

```diff
@@ -38,9 +38,10 @@ type Client struct {
 	pollDuration     time.Duration
 	cooldownDuration time.Duration
 
-	firstTimeGroup sync.Once
-	generated      uint32 // decide to wait if we have a generated url
-	matched        bool
+	firstTimeGroup   sync.Once
+	generated        uint32 // decide to wait if we have a generated url
+	matched          bool
+	matchedTemplates []string
```

Comment:
Intstead of a slice, we should use a fixed size LRU as done for other caches in interactsh.

Also, the map key can be set to a hash of template->target combo so it will guarantee unique results across templates.

Reply:
done